### ISSUE TITLE
optional or throws since 2.1.2

### ISSUE THREAD

**Issue by Rodsevich** [google/quiver-dart#584]
check this issue: https://github.com/dart-lang/angular_components/issues/445
We have to have quiver downgraded for working bc of this bug.

This is the function that in theory throws
https://github.com/google/quiver-dart/blob/f786fbf9f22beadd5d3e3f9723fe2cda9ef247d3/lib/src/core/optional.dart#L75-L80

And in the version before the function looks like this:

```dart
T or(T defaultValue) {
    if (defaultValue == null) {
      throw new ArgumentError('defaultValue must not be null.');
    }
    return _value == null ? defaultValue : _value;
}
```

It may be a sdk issue, maybe, but if you revert that piece of code you would simplify the life of us. Thanks

**Comment by cbracken** [comment]
Fixed in https://github.com/google/quiver-dart/pull/587

**Comment by cbracken** [comment]
Released in Quiver 2.1.3.
